apps/website-25/src/pages/api/courses/exercises/[exerciseId]/response.ts

```diff
@@ -0,0 +1,88 @@
+import { z } from 'zod';
+import createHttpError from 'http-errors';
+import db from '../../../../../lib/api/db';
+import { makeApiRoute } from '../../../../../lib/api/makeApiRoute';
+import {
+  ExerciseResponse,
+  exerciseResponseTable,
+} from '../../../../../lib/api/db/tables';
+
+export type GetExerciseResponseResponse = {
+  type: 'success',
+  exerciseResponse: ExerciseResponse,
+};
+
+export type PutExerciseResponseRequest = {
+  response: string,
+  completed?: boolean,
```

Comment:
I'm not sure why this should be optional - I'd expect all exercises should be able to have some sense of being completed or not?